src/platformdirs/macos.py

```diff
@@ -12,36 +33,43 @@ class MacOS(PlatformDirsABC):
     """
 
     @property
+    @_xdg_fallback
```

Comment:
You're trying to get too fancy here, don't use this complex annotation; just write the logic within the property body :+1: